### PR TITLE
South Carolina Two Wage Earner Credit

### PR DESCRIPTION
**Pull request by hua7450**
### Summary
📝📊🧪

This pull request adds a new feature to the policy engine: the South Carolina two wage earner credit. It defines the parameters, variables, and tests for the credit, and updates the changelog accordingly. The files added are `changelog_entry.yaml`, `earned_income.yaml`, `rate.yaml`, `sc_two_wage_earner_credit.yaml`, `sc_gross_earned_income.py`, and `sc_two_wage_earner_credit.py`.

> _`Two wage earner credit`_
> _New feature for South Carolina_
> _Added files and tests_

### Walkthrough
*  Add a new entry to the changelog file for the South Carolina two wage earner credit feature ([link](https://github.com/PolicyEngine/policyengine-us/pull/2608/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4))
*  Define the parameters for the credit, including the sources of earned income ([link](https://github.com/PolicyEngine/policyengine-us/pull/2608/files?diff=unified&w=0#diff-1f6512c22d189c2ebef96ff78649f02fc8af31be92b672c3175b82356d130985R1-R18)) and the rates and thresholds ([link](https://github.com/PolicyEngine/policyengine-us/pull/2608/files?diff=unified&w=0#diff-09dcd4f0d6ebd17c80819312eda96990dee80579f1b3f0a2005a045574f98dc3L1-R27))
*  Define the variables for the credit, including the gross earned income for each person (`sc_gross_earned_income`) ([link](https://github.com/PolicyEngine/policyengine-us/pull/2608/files?diff=unified&w=0#diff-a93b6cd19c64305b701d9a034b6fe07baa1bfbfc5b5bb425695d3dddf54678c0R1-R11)) and the credit amount for each tax unit (`sc_two_wage_earner_credit`) ([link](https://github.com/PolicyEngine/policyengine-us/pull/2608/files?diff=unified&w=0#diff-8a79c26d04f76eb9be0a2a266e27168eab6f8f13398bcd00cc43c424a24a53e6R1-R29))
*  Add four test cases for the credit in `sc_two_wage_earner_credit.yaml`, covering different scenarios of filing status, income, and credit amount ([link](https://github.com/PolicyEngine/policyengine-us/pull/2608/files?diff=unified&w=0#diff-3708189c6107871f39c65a01ca2cf40e59372067c06c8da21b1ded04bb02474eL1-R94))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/sc/tax/income/credits/two_wage_earner/rate.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+metadata:
+  reference:
+    - title: SC 1040 tax form instructions (2021)
+      href: https://dor.sc.gov/forms-site/Forms/IITPacket_2021.pdf#page=23
+    - title: SC 1040 tax form instructions (2022)
+      href: https://dor.sc.gov/forms-site/Forms/IITPacket_2022.pdf#page=23
+    - title: SC Legal Code |  Section 12-6-3330
+    # Legal code set it to $50,000, then amended it to $30,000 in 2017.
+    # Now adjusted for inflation.
+    - title: SC Legal Code |  Section 12-6-3330
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/sc/tax/income/credits/two_wage_earner/rate.yaml:23]: need href

File: policyengine_us/parameters/gov/states/sc/tax/income/credits/two_wage_earner/earned_income.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+description: South Carolina counts these income sources as earned income.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/sc/tax/income/credits/two_wage_earner/earned_income.yaml:1]:
```suggestion
description: South Carolina counts these income sources as earned income for the two wage earner credit.
```

File: policyengine_us/parameters/gov/states/sc/tax/income/credits/two_wage_earner/rate.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+description: South Carolina provides these rates under the two wage earner credit.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/sc/tax/income/credits/two_wage_earner/rate.yaml:1]:
```suggestion
description: South Carolina provides these rates of the filers earned income under the two wage earner credit.
```

File: policyengine_us/parameters/gov/states/sc/tax/income/credits/two_wage_earner/earned_income.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    - self_employment_income  
+    - qualified_business_income  
+    - partnership_s_corp_income 
+    - farm_income 
+    
+metadata:
+  unit: list
+  reference:
+    - title: Instructions - 2021 SC1040 - South Carolina Individual Income Tax Return
+      href: https://dor.sc.gov/forms-site/Forms/IITPacket_2021.pdf#page=22
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/sc/tax/income/credits/two_wage_earner/earned_income.yaml:16]: could you add 2022 tax form

File: policyengine_us/variables/gov/states/sc/tax/income/credits/two_wage_earner/sc_two_wage_earner_credit.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    defined_for = StateCode.SC
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        "https://dor.sc.gov/forms-site/Forms/SC1040TT_2021.pdf",
+        "https://dor.sc.gov/forms-site/Forms/IITPacket_2021.pdf#page=23",
+    )
+
+    def formula(tax_unit, period, parameters):
+        person = tax_unit.members
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/sc/tax/income/credits/two_wage_earner/sc_two_wage_earner_credit.py:17]: defined below
```suggestion
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**